src/core/tsi/transport_security.cc

```diff
@@ -213,10 +213,10 @@ tsi_result tsi_handshaker_next(
 
 void tsi_handshaker_shutdown(tsi_handshaker* self) {
   if (self == nullptr || self->vtable == nullptr) return;
-  self->handshake_shutdown = true;
   if (self->vtable->shutdown != nullptr) {
     self->vtable->shutdown(self);
   }
+  self->handshake_shutdown = true;
```

Comment:
Trying to figure out this part. It looked to me like the "shutdown no-op" was because of [this conditional within ALTS](https://github.com/grpc/grpc/blob/master/src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc#L357). Why make the change here rather than within ALTS?

Reply:
Good point. I think the conditional statement in ALTS does not have a problem, since we want to make the successive ALTS `handshaker_shutdown` to be no-op. It then requires the wrapper API to set the flag, which I think should also happen after the shutdown action takes place. Only ALTS has a need to implement `handshaker_shutdown` API, and for all other transport securities, the operation is no-op.